components/reusable-components/reusable-form/ReusableForm.tsx

```diff
@@ -0,0 +1,192 @@
+  const [password, setPassword] = useState<string>('');
+
+  const initialValues = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+  };
+
+  const validationSchema = Yup.object({
+    firstName: Yup.string()
+      .matches(/^[a-zA-Z .'-]+$/, '*Невалидно име!')
+      .min(2, '*Вашето име е премногу кратко!')
+      .max(50, '*Вашето име е премногу долго!')
+      .required('*Задолжително внесете име'),
+    lastName: Yup.string()
+      .min(2, '*Вашето презиме е премногу кратко!')
+      .max(50, '*Вашето презиме е премногу долго!')
+      .required('*Задолжително внесете презиме'),
+    email: Yup.string()
+      .email('*Невалидна емаил адреса')
+      .required('*Задолжително внесете електронка пошта'),
+    password: Yup.string()
+      .matches(/[a-zA-Z]/, '*Лозинката не ги исполнува сите услови!')
+      .min(8, '*Вашата лозинка е премногу кратка!')
+      .required('*Задолжително внесете лозинка'),
+  });
+
+  const formik: FormikProps<{
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+  }> = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: () => {},
+  });
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setPassword(value);
+    formik.handleChange(e);
+  };
+
+  const isPasswordValid = (rule: string, password: string): boolean => {
+    if (rule === 'uppercase') {
+      return /[A-Z]/.test(password);
+    }
+    if (rule === 'specialChar') {
+      return /[!@#$%^&*(),.?":{}|<>]/.test(password);
+    }
+    if (rule === 'minLength') {
+      return password.length >= 8;
+    }
+    return false;
+  };
+
+  return (
+    <div className={style.reusableFormWrapper}>
+      <h2 className={style.formTitle}>Регистрирај се</h2>
+      <form className={style.reusableForm} onSubmit={formik.handleSubmit}>
+        <TextInput
+          placeholder="Внесете го вашето име"
+          type="text"
+          label="Име *"
+          name="firstName"
+          field="firstName"
+          formik={formik}
+          inputClass={[`${style.signUpInput}`]}
+        />
+        <TextInput
+          placeholder="Внесете го вашето презиме"
+          type="text"
+          label="Презиме *"
+          name="lastName"
+          field="lastName"
+          formik={formik}
+          inputClass={[style.signUpInput]}
+        />
+        <TextInput
+          placeholder="Внесете ја вашата електронска пошта"
+          type="email"
+          label="Електронска пошта *"
+          name="email"
+          field="email"
+          formik={formik}
+          inputClass={['signUpInput']}
+        />
+        <TextInput
+          placeholder="Внесете ја вашата лозинка"
+          type="password"
+          label="Лозинка *"
+          name="password"
+          field="password"
+          formik={formik}
+          inputClass={['signUpInput']}
+          onChange={handlePasswordChange}
```

Comment:
First issue here is how we are handling onChange in the custom TextInput component, right now it handles only     `formik.handleChange(e);` but it should also handle the scenario where we are passing a onChange function like here